spec/std/enumerable_spec.cr

```diff
@@ -803,18 +803,51 @@ describe "Enumerable" do
 
   describe "max" do
     it { [1, 2, 3].max.should eq(3) }
+    it { [1, 2, 3].max(0).should eq([] of Int32) }
+    it { [1, 2, 3].max(1).should eq([3]) }
+    it { [1, 2, 3].max(2).should eq([3,2]) }
+    it { [1, 2, 3].max(3).should eq([3,2,1]) }
+    it { [1, 2, 3].max(4).should eq([3,2,1]) }
+    it { ([] of Int32).max(0).should eq([] of Int32) }
+    it { ([] of Int32).max(5).should eq([] of Int32) }
+    it {
+      (0..1000).map { |x| (x*137+x*x*139)%5000 }.max(10).should eq([
+        4992, 4990, 4980, 4972, 4962, 4962, 4960, 4960, 4952, 4952
+      ])
+    }
+
+    it "does not modify the array" do
+      xs = [7, 5, 2, 4, 9]
+      xs.max(2).should eq([9, 7])
+      xs.should eq([7, 5, 2, 4, 9])
+    end
 
     it "raises if empty" do
       expect_raises Enumerable::EmptyError do
         ([] of Int32).max
       end
     end
 
+    it "raises if n is negative" do
+      expect_raises ArgumentError do
+        ([1,2,3] of Int32).max(-1)
+      end
+      expect_raises ArgumentError do
+        ([1,2,3] of Int32).max(-2)
+      end
```

Comment:
These examples seem pretty similar to me... does it make sense to have both?

Considering edge cases, maybe an empty array would be a good addition? (`([] of Int32).max(-1)` )

Ditto for `#min` specs.